sw/applications/spi_flash_write/spi_flash_write.c

```diff
@@ -0,0 +1,274 @@
+    #endif
+
+    // dma peripheral structure to access the registers
+    dma_t dma;
+    dma.base_addr = mmio_region_from_addr((uintptr_t)DMA_START_ADDRESS);
+
+    soc_ctrl_t soc_ctrl;
+    soc_ctrl.base_addr = mmio_region_from_addr((uintptr_t)SOC_CTRL_START_ADDRESS);
+
+    // Enable interrupt on processor side
+    // Enable global interrupt for machine-level interrupts
+    CSR_SET_BITS(CSR_REG_MSTATUS, 0x8);
+    // Set mie.MEIE bit to one to enable machine-level fast spi interrupt
+    #ifndef USE_SPI_FLASH
+        const uint32_t mask = 1 << 20;
+    #else
+        const uint32_t mask = 1 << 21;
+    #endif
+    CSR_SET_BITS(CSR_REG_MIE, mask);
+    spi_intr_flag = 0;
+
+    // Select SPI host as SPI output
+    soc_ctrl_select_spi_host(&soc_ctrl);
+    // Enable SPI host device
+    spi_set_enable(&spi_host, true);
+
+    // SPI and SPI_HOST are the same IP so same register map
+    uint32_t *fifo_ptr = spi_host.base_addr.base + SPI_HOST_DATA_REG_OFFSET;
+
+    // -- SPI CONFIGURATION -- 
+    // Configure chip 0 (flash memory)
+    // Max 50 MHz core SPI clock --> Max 25 MHz SCK
+    // Single data IO; keep timing as safe as possible.
+    const uint32_t chip_cfg = spi_create_configopts((spi_configopts_t){
+        .clkdiv     = 1,
+        .csnidle    = 0xF,
+        .csntrail   = 0xF,
+        .csnlead    = 0xF,
+        .fullcyc    = false,
+        .cpha       = 0,
+        .cpol       = 0
+    });
+    spi_set_configopts(&spi_host, 0, chip_cfg);
+    spi_set_csid(&spi_host, 0);
+
+    /////////////// WRITE ////////////////
+
+    const uint32_t cmd_dummy = spi_create_command((spi_command_t){
+        .len        = 2,
+        .csaat      = false,
+        .speed      = kSpiSpeedStandard,
+        .direction  = kSpiDirTxOnly
+    });
+
+    const uint32_t cmd_dummy_read = spi_create_command((spi_command_t){
+        .len        = 2,
+        .csaat      = false,
+        .speed      = kSpiSpeedStandard,
+        .direction  = kSpiDirRxOnly
+    });
+
+    printf("Reset...\n");
+
+    // Reset
+    const uint32_t reset_cmd = 0xFFFFFFFF;
+    spi_write_word(&spi_host, reset_cmd);
+    const uint32_t cmd_reset = spi_create_command((spi_command_t){
+        .len        = 3,
+        .csaat      = false,
+        .speed      = kSpiSpeedStandard,
+        .direction  = kSpiDirTxOnly
+    });
+    spi_set_command(&spi_host, cmd_reset);
+    spi_wait_for_ready(&spi_host);
+    spi_set_rx_watermark(&spi_host,1);
+
+    printf("power up...\n");
+
+    // Power up flash
+    const uint32_t powerup_byte_cmd = 0xab;
+    spi_write_word(&spi_host, powerup_byte_cmd);
+    const uint32_t cmd_powerup = spi_create_command((spi_command_t){
+        .len        = 0,
+        .csaat      = false,
+        .speed      = kSpiSpeedStandard,
+        .direction  = kSpiDirTxOnly
+    });
+    spi_set_command(&spi_host, cmd_powerup);
+    spi_wait_for_ready(&spi_host);
+    printf("Wait...\n");
```

Comment:
Consider removing printf that do not provide so much information and slow down the simulation